client/core/core.go

```diff
@@ -2751,23 +2756,23 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 	err = dc.signAndRequest(msgOrder, route, result, fundingTxWait+DefaultResponseTimeout)
 	if err != nil {
 		unlockCoins()
+		// At this point there is a possibility that the server got the request
+		// and created the trade order, but we lost the connection before
+		// receiving the response with the trade's order ID. Any preimage
+		// request will be unrecognized. This order is ABANDONED.
 		return nil, 0, fmt.Errorf("new order request with DEX server %v market %v failed: %w", dc.acct.host, mktID, err)
 	}
 
 	// If we encounter an error, perform some basic logging.
-	//
-	// TODO: Notify the client somehow.
 	logAbandon := func(err interface{}) {
-		c.log.Errorf("Abandoning order. "+
-			"preimage: %x, server time: %d: %v",
+		c.log.Errorf("Abandoning order. preimage: %x, server time: %d: %v",
 			preImg[:], result.ServerTime, err)
 	}
 
-	err = validateOrderResponse(dc, result, ord, msgOrder)
+	err = validateOrderResponse(dc, result, ord, msgOrder) // stamps the order, giving it a valid ID
```

Comment:
You can manually test the async handling by sleeping for a few seconds here, allowing server to send a preimage request if the epoch closes while sleeping but before we put it in our trades map.  Works as expected. e.g.:

> (order placed and sleeping)
2021-03-09 08:47:45.001 [WRN] CORE: Received preimage request for d20502da753eb5912ea984efa69a2c3b5f8ec9d8e7f89491eac772a24900684b before we got an order submission response!
(still sleeping with nothing blocking, then awake and signalling on syncChan)
> 2021-03-09 08:47:46.911 [DBG] CORE: notify: |POKE| (order) Order placed - selling 10.00000000 dcr, rate = 0.10000000 (d20502da) - Order: d20502da753eb5912ea984efa69a2c3b5f8ec9d8e7f89491eac772a24900684b
> 2021-03-09 08:47:46.911 [TRC] CORE: notify: |DATA| (order) preimage sent - Order: d20502da753eb5912ea984efa69a2c3b5f8ec9d8e7f89491eac772a24900684b
> 2021-03-09 08:47:46.911 [DBG] CORE: async processPreimageRequest for d20502da753eb5912ea984efa69a2c3b5f8ec9d8e7f89491eac772a24900684b succeeded
> 2021-03-09 08:47:46.912 [INF] CORE: Standing order d20502da did not match and is now booked.